evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -135,6 +135,47 @@ def test_iterative_algorithm_results(ensembling_value, dummy_binary_pipeline_cla
             assert pipeline.model_family == ModelFamily.ENSEMBLE
 
 
+@pytest.mark.parametrize("ensembling_value", [True, False])
+def test_iterative_algorithm_one_allowed_pipeline(ensembling_value, logistic_regression_binary_pipeline_class):
+    # Checks that when len(allowed_pipeline) == 1, ensembling is not run, even if set to True
+    algo = IterativeAlgorithm(allowed_pipelines=[logistic_regression_binary_pipeline_class], ensembling=ensembling_value)
+    assert algo.pipeline_number == 0
+    assert algo.batch_number == 0
+    assert algo.allowed_pipelines == [logistic_regression_binary_pipeline_class]
+
+    # initial batch contains one of each pipeline, with default parameters
+    next_batch = algo.next_batch()
+    assert len(next_batch) == 1
+    assert [p.__class__ for p in next_batch] == [logistic_regression_binary_pipeline_class] * len(next_batch)
+    assert algo.pipeline_number == 1
+    assert algo.batch_number == 1
+    assert all([p.parameters == (p.__class__)({}).parameters for p in next_batch])
```

Comment:
Nit-pick: I think we can use `p.default_parameters` or `p.__class__.default_parameters` here to make the intention clearer.

I think we can also do something similar on line 176.